test/integration/install_test.go

```diff
@@ -679,6 +693,138 @@ func TestRetrieveUidPostUpgrade(t *testing.T) {
 	}
 }
 
+func TestOverridesSecret(t *testing.T) {
+
+	if TestHelper.GetHelmReleaseName() != "" {
+		t.Skip("Skipping as this is a helm test where linkerd-config-overrides is absent")
+	}
+
+	configOverridesSecret, err := TestHelper.KubernetesHelper.GetSecret(context.Background(), TestHelper.GetLinkerdNamespace(), "linkerd-config-overrides")
+	if err != nil {
+		testutil.AnnotatedFatalf(t, "could not retrieve linkerd-config-overrides",
+			"could not retrieve linkerd-config-overrides\n%s", err.Error())
+	}
+
+	overrides := configOverridesSecret.Data["linkerd-config-overrides"]
+	overridesTree, err := tree.BytesToTree(overrides)
+	if err != nil {
+		testutil.AnnotatedFatalf(t, "could not retrieve linkerd-config-overrides",
+			"could not retrieve linkerd-config-overrides\n%s", err.Error())
+	}
+
+	// Check for fields that were added during install
+	testCases := []struct {
+		tree  tree.Tree
+		path  []string
+		value string
+	}{
+		{
+			overridesTree,
+			[]string{"controllerLogLevel"},
+			"debug",
+		},
+		{
+			overridesTree,
+			[]string{"proxyInit", "ignoreInboundPorts"},
+			"1234,5678",
+		},
+	}
+
+	// Check for fields that were added during upgrade
+	if TestHelper.UpgradeFromVersion() != "" {
+		testCases = append(testCases, []struct {
+			tree  tree.Tree
+			path  []string
+			value string
+		}{
+			{
+				overridesTree,
+				[]string{"proxyInit", "ignoreOutboundPorts"},
+				"1234,5678",
+			},
+		}...)
+	}
+
+	for _, tc := range testCases {
+		tc := tc // pin
+		t.Run(fmt.Sprintf("%s: %s", strings.Join(tc.path, "/"), tc.value), func(t *testing.T) {
+			finalValue, err := tc.tree.GetString(tc.path...)
+			if err != nil {
+				testutil.AnnotatedFatalf(t, "could not perform tree.GetString",
+					"could not perform tree.GetString\n%s", err.Error())
+			}
+
+			if tc.value != finalValue {
+				testutil.AnnotatedFatalf(t, fmt.Sprintf("Values at path %s do not match", strings.Join(tc.path, "/")),
+					"Expected value at [%s] to be [%s] but received [%s]",
+					strings.Join(tc.path, "/"), tc.value, finalValue)
+			}
+		})
+	}
+
+	panicIfError := func(t *testing.T, f func(...string) (string, error), path ...string) string {
+		val, err := f(path...)
+		if err != nil {
+			testutil.AnnotatedFatalf(t, err.Error(),
+				err.Error(), err.Error())
+			return ""
+
+		}
+		return val
+	}
+
+	t.Run("Check if any unknown fields sneaked in", func(t *testing.T) {
+		knownKeys := tree.Tree{
+			"controllerLogLevel": "debug",
+			"identity": map[string]interface{}{
+				"issuer": map[string]interface{}{},
+			},
+			"identityTrustAnchorsPEM": panicIfError(t, overridesTree.GetString, []string{"identityTrustAnchorsPEM"}...),
```

Comment:
In this call and all the others below, you can pass a list of args instead of having to build a slice.